lib/findPattern.ts

```diff
@@ -346,8 +349,11 @@ async function findPatternInZip(
 
   // Get file size
   const headResp = await fetch(url, {
-    method: "HEAD",
-    headers: { "Accept-Encoding": "identity" },
+    method: options.useGetCheck ? "GET" : "HEAD",
+    headers: {
+      "Accept-Encoding": "identity",
+      ...(options.useGetCheck ? { "Range": "bytes=0-0" } : {}),
```

Comment:
The trick is to use an empty `Range` to just get the headers, which makes it more or less equivalent to a `HEAD` request. Note that I didn’t make it the default because some servers may not support the `Range` header.